source/slang/slang-emit-cpp.cpp

```diff
@@ -2589,34 +2589,43 @@ void CPPSourceEmitter::emitModuleImpl(IRModule* module)
                             default:
 
                                 auto targetProgram = program->getTargetProgram(targetRequest);
-                                DiagnosticSink sink(linkage->getSourceManager());
+                                //DiagnosticSink sink(linkage->getSourceManager());
                                 CompileResult result =
-                                    targetProgram->getOrCreateEntryPointResult(index, &sink);
+                                    targetProgram->getOrCreateEntryPointResult(index, sink);
 
                                 Slang::ComPtr<ISlangBlob> blob;
                                 if (SLANG_FAILED(result.getBlob(blob)))
                                 {
-                                    SLANG_UNEXPECTED("No blob to emit");
-                                    return;
+                                    StringBuilder builder;
+                                    sink->diagnoseRaw(Severity::Error, UnownedStringSlice
```

Comment:
The use of UnownedStringSlice ctor works, but internally does a strlen. If you wanted to avoid that, you can do 

```
UnownedStringSlice::fromLiteral("Some literal");
```

And it doesn't need to do the strlen - it can compile time determine length. There's not much in this I'll grant you - but it seems that there is already a diagnoseRaw overload that just takes a const char*, which is slightly less unsightly to invoke.